src/subject-metadata/SubjectMetadataController.ts

```diff
@@ -13,16 +13,30 @@ const controllerName = 'SubjectMetadataController';
 
 type SubjectOrigin = string;
 
+/**
+ * The different kinds of subjects that MetaMask may interact with, including
+ * third parties and itself (e.g., when the background communicated with the UI).
+ */
+export enum SubjectType {
+  Extension = 'extension',
+  Internal = 'internal',
+  Unknown = 'unknown',
+  Website = 'website',
+  Snap = 'snap',
+}
+
 export type SubjectMetadata = PermissionSubjectMetadata & {
   [key: string]: Json;
   // TODO:TS4.4 make optional
   name: string | null;
+  subjectType: SubjectType | null;
```

Comment:
Just making sure I understand, this is not a breaking change because we already specify this via `[key: string]: Json;` in the extension?

Reply:
Yes, and it's optional in `SubjectMetadataToAdd`.